エンジンビルド後のテストがmacOSで落ちる

## 不具合の内容

エンジンをビルドした後に自動で走るrelease-test workflowがmacOSでだけ落ちてたのでとりあえずissueを作りました。
実際にエンジンをダウンロードして使ってみた感じはまともに動きました。

ちゃんと調べてなくてただの予想ですが、github actionsのmacosのスペックが縮小されたんじゃないかなと思ってます。
最近macos-latestが更新されたあたりにエディタ側もスペック不足が原因っぽいエラーがでてたので･･･。
https://github.com/VOICEVOX/voicevox/issues/2007

### 現象・ログ

Github Workflowの結果↓
https://github.com/VOICEVOX/voicevox_engine/actions/runs/8852174690

ログ↓
```log
Run python build_util/check_release_build.py --dist_dir dist/
Warning: cpu_num_threads is set to 0. Setting it to half of the logical cores.
INFO:     Started server process [3165]
INFO:     Waiting for application startup.
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 1348, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 1283, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 1329, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 1278, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 1038, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 976, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 942, in connect
    self.sock = self._create_connection(
                ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/socket.py", line 851, in create_connection
    raise exceptions[0]
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/socket.py", line 836, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/runner/work/voicevox_engine/voicevox_engine/build_util/check_release_build.py", line 72, in <module>
    test_release_build(dist_dir=args.dist_dir, skip_run_process=args.skip_run_process)
  File "/Users/runner/work/voicevox_engine/voicevox_engine/build_util/check_release_build.py", line 32, in test_release_build
    with urlopen(req) as res:
         ^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 519, in open
    response = self._open(req, data)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 496, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 1377, in http_open
    return self.do_open(http.client.HTTPConnection, req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 1351, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 61] Connection refused>
Error: Process completed with exit code 1.
```
## その他

とりあえずエンジンの起動待機時間を長くすれば解決できるのではと楽観的に考えてます。

--- a/.github/workflows/build-engine-package.yml
+++ b/.github/workflows/build-engine-package.yml
@@ -77,7 +77,7 @@ jobs:
             zlib_url: http://www.winimage.com/zLibDll/zlib123dllx64.zip
             target: windows-nvidia
           # Mac CPU (x64 arch only)
-          - os: macos-11
+          - os: macos-12
             architecture: "x64"
             voicevox_core_asset_prefix: voicevox_core-osx-x64-cpu
             onnxruntime_url: https://github.com/microsoft/onnxruntime/releases/download/v1.13.1/onnxruntime-osx-x86_64-1.13.1.tgz

--- a/.github/workflows/test-engine-package.yml
+++ b/.github/workflows/test-engine-package.yml
@@ -41,7 +41,7 @@ jobs:
             target: linux-cpu
           - os: ubuntu-20.04
             target: linux-nvidia
-          - os: macos-11
+          - os: macos-12
             target: macos-x64
           - os: windows-2019
             target: windows-cpu

